Add the parameter ssl_protocols to specify the SSL protocols

lei-tang opened the pull request:
Add the parameter ssl_protocols to specify the SSL protocols. The "ssl_protocols" input parameter can be repeated multiple times to specify multiple SSL protocols,  e.g., --ssl_protocols=TLSv1.1 --ssl_protocols=TLSv1.2, which only allows TLS 1.1 and 1.2 while disabling TLS 1.0.
Refer to https://nginx.org/en/docs/http/ngx_http_ssl_module.html#ssl_protocols.

lei-tang committed:
Add the parameter ssl_protocols to specify the SSL protocols

Add the parameter ssl_protocols to specify the SSL protocols.
For example, --ssl_protocols="TLSv1.1 TLSv1.2" to only allow TLS 1.1 and
1.2 while disabling TLS 1.0.
Refer to  https://nginx.org/en/docs/http/ngx_http_ssl_module.html#ssl_protocols.

qiwzhang commented on start_esp/nginx-auto.conf.template:
```diff
@@ -64,6 +64,8 @@ http {
   server {
     server_name ${ingress.host};
 
+    ssl_protocols ${ssl_protocols};
```
Prefer to not set ssl_protocols as default.

lei-tang replied:
Done: changed to None.

qiwzhang commented on start_esp/start_esp.py:
```diff
@@ -696,6 +700,13 @@ def make_argparser():
         still be enabled from request HTTP headers with trace context regardless
         this flag value.
         ''')
+    parser.add_argument('--ssl_protocols',
```
I have some concerns about flag value with space.   Is it possible that ssl_protocols can be specified multiple times, so it can be used to specify multiple versions, such as

--ssl_protocols=TLSv1.1  --ssl_protocols=TLSv1.2

lei-tang replied:
Done.

lei-tang committed:
Revised based on the review comments

- Change the "ssl_protocols" default value to None.
- The "ssl_protocols" argument can be repeat multiple times to specify multiple SSL protocols
(e.g., --ssl_protocols=TLSv1.1 --ssl_protocols=TLSv1.2).

lei-tang committed:
Update the ssl_protocol command manual

lei-tang committed:
Fix a typo
